# HG changeset patch
# User Ray Donnelly <mingw.android@gmail.com>
# Date 1471253669 -3600
#      Mon Aug 15 10:34:29 2016 +0100
# Node ID b9b4c840848ccb1b01fe1f63039646a7e0acfc16
# Parent  5b0b4f32f6f9e1c0a6d023dc6d066474bd87d2de
Win32: Use get_R_HOME() if R_HOME is not set.

This will defer to the loaded R.dll to get R_HOME which
is also what RStudio does.

diff --git a/rpy/rinterface/_rinterface.c b/rpy/rinterface/_rinterface.c
--- a/rpy/rinterface/_rinterface.c
+++ b/rpy/rinterface/_rinterface.c
@@ -1252,6 +1252,8 @@
   R_DefParams(Rp);
   if (getenv("R_HOME")) {
     strcpy(RHome, getenv("R_HOME"));
+  } else if (get_R_HOME()) {
+    strcpy(RHome, get_R_HOME());
   } else {
     PyErr_Format(PyExc_RuntimeError, "R_HOME not defined.");
     return NULL;
